### ISSUE TITLE
imagepng() quality

### ISSUE THREAD

**Issue by bloxcms** [claviska/SimpleImage#252]
Replace

```php
imagepng($this->image, null, round(9 * $quality / 100));
```

by

```php
imagepng($this->image, null, round(9*(1-$quality/100))); 
```

The third param in `imagepng()` is not quality but compression 0-9

**Comment by claviska** [comment]
You're right, that parameter doesn't make much sense for PNG images. Even if "quality" somehow correlated to compression, it's currently reversed.

I'm voting to remove quality from PNG and use the default zlib compression value (same as omitting the `$quality` argument). This will affect the size of PNG images, but I don't consider it a breaking change since PNG is a lossless format.

Any thoughts or objections to this?

### TODO

- [ ] Remove "quality" argument from `imagepng`
- [ ] Update docs to state that quality only affects JPEG images since GIF and PNG are lossless formats

**Comment by bloxcms** [comment]
GD does not compress large files over 1000px.Do you plan to migrate to the PHP's Imagick?  Владислав АливановВеб-студия "Блокс", Набережные Челныт. +7-987-000-44-00, vlad@blox.ru    08.04.2020, 16:57, "Cory LaViska" <notifications@github.com>: You're right, that parameter doesn't make much sense for PNG images. Even if "quality" somehow correlated to compression, it's currently reversed.I'm voting to remove quality from PNG and use the default zlib compression value (same as omitting the $quality argument). This will affect the size of PNG images, but I don't consider it a breaking change since PNG is a lossless format.Any thoughts or objections to this?TODORemove "quality" argument from imagepngUpdate docs to state that quality only affects JPEG images since GIF and PNG are lossless formats—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by claviska** [comment]
No plans to use Imagick, as ImageMagick is not installed by default on most systems and is often a pain point for those who try to install it (especially on shared hosting environments).

As far as I'm aware, there's no explicit limit to the size of an image GD will process. This is usually limited by available memory. [See this post](https://stackoverflow.com/questions/3381228/php-gd-imagecreatefromjpeg-cant-handle-big-size-image/3381254) for more info and a possible solution.

**Comment by claviska** [comment]
Fixed in 22336072d501c210d7e950af44b0fec9e4ed0a73
